### PR TITLE
Add Haiku OS cross-compile support

### PR DESCRIPTION
**Pull request by erysdren**
x86 and x86_64 both tested and working in QEMU and on bare metal.

[Image, running in QEMU.](https://erysdren.me/files/quakespasm_haiku.png)

### CODE REVIEW

File: Quake/sys_sdl_unix.c [hunk 1]
```diff
@@ -321,6 +321,21 @@ static void Sys_GetBasedir (char *argv0, char *dst, size_t dstsize)
...
+		perror("realpath");
+		if (getcwd(dst, dstsize - 1) == NULL)
+	_fail:		Sys_Error ("Couldn't determine current directory");
+	}
+	else
+	{
+		/* strip off the binary name */
+		if (! (tmp = strdup (dst))) goto _fail;
+		q_strlcpy (dst, dirname(tmp), dstsize);
+		free (tmp);
```
Comment (sezero) [inline Quake/sys_sdl_unix.c:336]: How are you stripping off the binary name here??

Comment (erysdren) [reply Quake/sys_sdl_unix.c:336]: To be completely honest with you, I just copied that code from the OSX version of Sys_GetBasedir() at line 309 in that file. I don't know why it works.

Comment (sezero) [reply Quake/sys_sdl_unix.c:336]: Agh.. Didn't notice that you're passing `dirname(tmp)` to strlcpy. Sorry for the noise.

Will pull this in, soon

**Comment by sezero** [comment]
This is in now. Thanks!

**Pull request merged**